jerry-core/include/jerryscript-port.h

```diff
@@ -20,6 +20,21 @@
 #include <stdint.h>
 #include <stdio.h>
 
+#ifdef _MSC_VER
+void * __cdecl _alloca (size_t _Size);
+#define JERRY_ALLOCA(name, type, size) type *name = (type *)(_alloca (sizeof(type)*size))
+#else
+#define JERRY_ALLOCA(name, type, size) type name[size]
+#endif
+
+#ifdef _MSC_VER
+#define __noreturn __declspec(noreturn)
+#define __attr_format___(...)
+#else
+#define __noreturn __attribute__ ((noreturn))
+#define __attr_format___(...) __attribute__ ((format (__VA_ARGS__)))
+#endif
```

Comment:
FYI: I've proposed a solution to handle the pollution of -port.h in #2313 . Should that be accepted, then these compiler-specific definitions could move to the new header. (Because as it was discussed, they simply don't look right here. Especially `JERRY_ALLOCA`, as it kind of "abuses" jerryscript-port.h to make sure that the macro is visible everywhere.)